lms/djangoapps/grades/tests/test_new.py

```diff
@@ -230,3 +236,149 @@ def test_save_and_load(self):
+            category=u'vertical',
+            display_name=u'Test Vertical 1'
+        )
+
+    def _get_fresh_subsection_score(self, course_structure, subsection):
+        """
+        Return a Score object for the specified subsection.
+
+        Ensures that a stale cached value is not returned.
+        """
+        subsection_factory = SubsectionGradeFactory(
+            self.student,
+            course_structure=course_structure,
+            course=self.course,
+        )
+        return subsection_factory.update(subsection)
+
+    def _get_altered_metadata(self, alterations):
+        """
+        Returns a copy of the default_problem_metadata dict updated with the
+        specified alterations.
+        """
+        metadata = self.default_problem_metadata.copy()
+        metadata.update(alterations)
+        return metadata
+
+    def _get_score_with_alterations(self, alterations):
+        """
+        Given a dict of alterations to the default_problem_metadata, return
+        the score when one correct problem (out of two) is submitted.
+        """
+        metadata = self._get_altered_metadata(alterations)
+
+        add_xml_block_from_file(u'problem', u'capa.xml', parent=self.vert1, metadata=metadata)
+        course_structure = get_course_blocks(self.student, self.course.location)
+
+        self.submit_question_answer(u'problem', {u'2_1': u'Correct'})
+        return self._get_fresh_subsection_score(course_structure, self.seq1)
+
+    def test_score_submission_for_capa_problems(self):
+        add_xml_block_from_file(u'problem', u'capa.xml', parent=self.vert1, metadata=self.default_problem_metadata)
+        course_structure = get_course_blocks(self.student, self.course.location)
+
+        score = self._get_fresh_subsection_score(course_structure, self.seq1)
+        self.assertEqual(score.all_total.earned, 0.0)
+        self.assertEqual(score.all_total.possible, 2.5)
+
+        self.submit_question_answer(u'problem', {u'2_1': u'Correct'})
+        score = self._get_fresh_subsection_score(course_structure, self.seq1)
+        self.assertEqual(score.all_total.earned, 1.25)
+        self.assertEqual(score.all_total.possible, 2.5)
+
+    @ddt.data(
+        (u'openassessment', u'openassessment.xml'),
+        (u'coderesponse', u'coderesponse.xml'),
+        (u'lti', u'lti.xml'),
+        (u'library_content', u'library_content.xml'),
+    )
+    @ddt.unpack
+    def test_loading_different_problem_types(self, block_type, filename):
+        """
+        Test that transformation works for various block types
+        """
+        if block_type == u'library_content':
+            # Library content does not have a weight
+            metadata = self._get_altered_metadata({u'graded': True})
+        else:
+            metadata = self.default_problem_metadata
+        add_xml_block_from_file(block_type, filename, parent=self.vert1, metadata=metadata)
+
+    @ddt.data(
+        ({}, 1.25, 2.5),
+        ({u'weight': 27}, 13.5, 27),
```

Comment:
I found I was using numbers like `2`, because they were simple, and they were potentially hiding issues, because a weight of two and a two part problem could lead to mistaking one doubling for the other.  27 seemed unlikely to be confused for something else.